packages/api/src/controllers/transcode.ts

```diff
@@ -0,0 +1,54 @@
+import { authorizer, validatePost } from "../middleware";
+import { Router } from "express";
+import { TranscodePayload } from "../schema/types";
+import { toObjectStoreUrl } from "./helpers";
+import { ensureQueueCapacity } from "../task/scheduler";
+
+const app = Router();
+
+app.post(
+  "/",
+  authorizer({}),
+  validatePost("transcode-payload"),
+  async (req, res) => {
+    const params = req.body as TranscodePayload;
+    const { catalystPipelineStrategy = undefined } = req.user.admin
+      ? params
+      : {};
+
+    await ensureQueueCapacity(req.config, req.user.id);
+
+    const taskType = "transcode-file";
+    let inUrl = params.input.url;
+    if (!inUrl) {
+      if (!params.input.path) {
+        throw new Error("Undefined property 'input.path'");
+      }
+      inUrl = toObjectStoreUrl(params.input) + params.input.path;
+    }
+    const storageUrl = toObjectStoreUrl(params.storage);
+
+    const task = await req.taskScheduler.spawnTask(
+      taskType,
+      {
+        [taskType]: {
+          input: {
+            url: inUrl,
+          },
+          storage: {
+            url: storageUrl,
+          },
+          outputs: params.outputs,
+          catalystPipelineStrategy,
+        },
+      },
+      null,
+      null,
+      req.user.id
+    );
+    await req.taskScheduler.enqueueTask(task);
+    res.json({ task: { id: task.id } });
```

Comment:
Actually, the only reason we didn't return the full task object in the response to the upload requests was that because, on that API, the task object is supposed to be a "background"/"power-user" kind of thing, while most users should only be dealing with the asset objects instead. 

On this new API though, the task is pretty much the core concept that the developer will be dealing with, and so is not something we should hide in the API responses either. So my suggestion would be to return the full task object on the API response here, instead of only the ID of such task which creates an annoying indirection.

WDYT? Also a note that if we follow that path, need to also make sure we clean up the response task object on this API here (although it's less critical to return the credentials to the same request that created the task anyway).

Reply:
I think it's a good idea. Updated to return the task object.